d3-selection/index.d.ts

```diff
@@ -12,7 +14,7 @@
  * without 'd3-selection' trying to use properties internally which would otherwise not
  * be supported.
  */
-export type BaseType = Element | EnterElement | Window;
+export type BaseType = Element | EnterElement | Document | Window | null;
```

Comment:
I'm still not convinced we need `| null` in `BaseType`, since selections will always ignore them.